spec/models/firm_factory_spec.rb

```diff
@@ -0,0 +1,189 @@
+RSpec.describe 'Firm factory' do
+  subject do
+    begin
+      FactoryGirl.create(factory)
+    rescue ActiveRecord::RecordInvalid
```

Comment:
@asmega this is a really good point. I deliberated about this a little. The reason I chose to do it this way is, we can test what really happens rather than forcing it one way or another. At the point I implemented this replacing the default subject `create` with a `build` showed up no breakages.